src/Core/Checkout/Promotion/Subscriber/PromotionIndividualCodeRedeemer.php

```diff
@@ -49,9 +74,9 @@ public function onOrderPlaced(CheckoutOrderPlacedEvent $event): void
             $code = $item->getPayload()['code'] ?? '';
 
             try {
-                // first try if its an individual
+                // first try if it's an individual
                 // if not, then it might be a global promotion
-                $individualCode = $this->getIndividualCode($code, $event->getContext());
+                $individualCode = $this->getIndividualCode($code, $context);
```

Comment:
I'm sorry for being nitpicky, but I really don't like this being called in a loop. Now, that we are not just doing this while placing an order (which is a one time event per request), this can be called thousands of times, if an external source syncs against Shopware.

Could we maybe do multiple loops, collecting the codes?